src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx

```diff
@@ -1,42 +1,137 @@
-    ): JSX.Element => (
-      <AnnotationTicketRow
-        issueKey={issueKey}
-        url={url}
-        source={source}
-        jiraTicket={jiraTicket}
-      />
-    ),
-  },
-];
+export const AnnotationTicketsTable: React.FC<Props> = ({
+  taskId,
+  execution,
+  userModifyPermission,
+  jiraIssues,
+  isIssue,
+}) => {
+  const dispatchBanner = useBannerDispatchContext();
+  const columns = [
+    {
+      render: (
+        text: string,
+        { issueKey, url, source, jiraTicket }: AnnotationTickets[0]
+      ): JSX.Element => (
+        <AnnotationTicketRow
+          issueKey={issueKey}
+          url={url}
+          source={source}
+          jiraTicket={jiraTicket}
+        />
+      ),
+    },
+    {
+      title: "Actions",
+      render: (
+        text: string,
+        { issueKey, url }: AnnotationTickets[0]
+      ): JSX.Element => (
+        <ConditionalWrapper
+          condition={!userModifyPermission}
+          wrapper={(children) => (
+            <Tooltip title="You are not authorized to edit task annotations">
+              <span>{children}</span>
+            </Tooltip>
+          )}
+        >
+          <BtnContainer>
+            {/* todo: add move button here */}
+            <Popconfirm
+              icon={null}
+              placement="topRight"
+              title="Delete this ticket?"
+              onConfirm={() => {
+                onClickRemove(url, issueKey);
+              }}
+              okText="Yes"
+              cancelText="Cancel"
+            >
+              <StyledButton
+                size="small"
+                data-cy="delete-btn"
+                glyph={<Icon glyph="Trash" />}
+                disabled={!userModifyPermission}
+              />
+            </Popconfirm>
+          </BtnContainer>
+        </ConditionalWrapper>
+      ),
+    },
+  ];
+  // todo: is loading useful or necessary?
+  const [removeAnnotation] = useMutation<
+    RemoveAnnotationIssueMutation,
+    RemoveAnnotationIssueMutationVariables
+  >(REMOVE_ANNOTATION, {
+    onCompleted: () => {
+      dispatchBanner.successBanner(`Successfully removed issue`);
+    },
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error removing the issue: ${error.message}`
+      );
+    },
+    refetchQueries: ["GetTask"],
+  });
 
-export const AnnotationTicketsTable: React.FC<{
-  jiraIssues: AnnotationTickets;
-}> = ({ jiraIssues }) => (
-  <TableWrapper>
-    <Table
-      data-test-id="build-baron-table"
-      dataSource={jiraIssues}
-      rowKey={({ issueKey }) => issueKey}
-      columns={columns}
-      pagination={false}
-      showHeader={false}
-    />
-  </TableWrapper>
-);
+  const onClickRemove = (url, issueKey) => {
+    const apiIssue = {
+      url,
+      issueKey,
+    };
+    removeAnnotation({ variables: { taskId, execution, apiIssue, isIssue } });
+  };
+
+  return (
+    <TableWrapper>
+      <ErrorBoundary>
```

Comment:
can you remind me of the error we're hoping to catch?

Reply:
replied on slack

Reply:
Im also curious

Reply:
I'll send you a gif on slack.